crates/json/src/lib.rs

```diff
@@ -0,0 +1,41 @@
+use std::collections::BTreeMap;
+use std::ops::{Deref, DerefMut};
+
+use drawbridge_hash::Hash;
+use serde::{Deserialize, Serialize};
+
+/// A directory
+#[derive(Clone, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Directory(BTreeMap<String, Entry>);
+
+impl Directory {
+    pub const TYPE: &'static str = "application/vnd.drawbridge.directory.v1+json";
+}
+
+impl Deref for Directory {
+    type Target = BTreeMap<String, Entry>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for Directory {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+/// A directory entry
+///
+/// Note that this type is designed to be extensible. Therefore, the fields
+/// here represent the minimum required fields. Other fields may be present.
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Entry {
+    /// The hash of this entry
+    pub hash: Hash,
```

Comment:
No `from` field as shown in the example json?

Reply:
No. As the comment says, this structure represents only the **minimum** required fields. If you want to enforce semantic validation of other fields it should probably happen in other layers above.